ibis/backends/postgres/__init__.py

```diff
@@ -311,28 +351,62 @@ def list_tables(
         with self._safe_raw_sql(sql) as cur:
             out = cur.fetchall()
 
+        # Include temporary tables only if no database has been explicitly specified
+        # to avoid temp tables showing up in all calls to `list_tables`
+        if db == "public":
+            out += self._fetch_temp_tables()
+
         return self._filter_with_like(map(itemgetter(0), out), like)
 
-    def list_databases(self, like=None) -> list[str]:
+    def _fetch_temp_tables(self):
+        # postgres temporary tables are stored in a separate schema
+        # so we need to independently grab them and return them along with
+        # the existing results
+
+        sql = (
+            sg.select("table_name")
+            .from_(sg.table("tables", db="information_schema"))
+            .distinct()
+            .where(C.table_type.eq("LOCAL TEMPORARY"))
+            .sql(self.dialect)
+        )
+
+        with self._safe_raw_sql(sql) as cur:
+            out = cur.fetchall()
+
+        return out
```

Comment:
both `risingwave` and `postgres` were returning all tables in all databases (formerly `schema`) when any `list_tables` call was made.

I fixed that to only return tables in `self.current_database` but that leaves out temp tables, so I added this helper to also return temp tables (which live in their own set of postgres-schema that have unpredictable names).